R/path-explorer.R

```diff
@@ -0,0 +1,73 @@
+
+addBranch <- function(originalPathString, arrayToGo) {
+  # Function to create a multilayered list called endList with a branched structure corresponding to the structure of arrayToGo that terminates with a string called 'path' that is equal to the string originalString
+  if (length(arrayToGo) == 0) {
+    # If stringToGo is empty, create a terminal list with a string called 'path' and value equal to originalString
+    endList <- list()
+    endList$path <- originalPathString
+    return(endList)
+  }
+  else {
+    # If arrayToGo is still not empty, remove its leading element and create a sub-branch list corresponding to the structure of the remaining elements of arrayToGo
+    newBranch <- list()
+    newBranch[[arrayToGo[1]]] <- addBranch(originalPathString, tail(arrayToGo, -1))
+    return(newBranch)
+  }
+}
+
+nextStep <- function(listSoFar, originalString, arrayToGo) {
+  # Recursive function that adds a multilayer list to listSoFar that has a branched structure representing the vector of strings stringToGo.
+  if (length(arrayToGo) == 0) { # If end of string vector stringToGo has been reached, create a vector called 'path' and give it the value 'originalString'.
+    listSoFar$path <- originalString
+  }
+  else { # End of branch has not been reached.
+    # If this portion of the string vector stringToGo has not been added to listToGo yet, add it using the function addBranch
+    if (is.null(listSoFar[[arrayToGo[1]]])) {
+      listSoFar[[arrayToGo[1]]] <- addBranch(originalString, tail(arrayToGo, -1))
+    }
+    # If this portion of the string vector stringToGo has already been added to listSoFar, remove the leading element of stringToGo and recursively apply this function using the remaining elements of stringToGo.
+    else {
+      listSoFar[[arrayToGo[1]]] <- nextStep(listSoFar[[arrayToGo[1]]], originalString, tail(arrayToGo, -1))
+    }
+  }
+  return(listSoFar)
+}
+
+#'  Given a simulation file path or an instance of a simulation, traverses the simulation structure and returns a tree like structure
+#'  allowing for intuitive navigration in the simulation tree
+#
+#' @param simulationOrFilePath Full path of the simulation to load or instance of a simulation
+#' @return A list with a branched structure representing the path tree of Quantities in the simulation file.
+#' At the end of each branch is a string called 'path' that is the path of the quantity represented by the branch.
+#'
+#' @examples
+#' simPath <- system.file("extdata", "simple.pkml", package = "ospsuite")
+#' sim <- loadSimulation(simPath)
+#'
+#' tree <- exploreSimulation(sim)
+#'
+#' liver_volume_path <- tree$Organism$Liver$Volume$path
+#'
+#' @export
+exploreSimulation <- function(simulationOrFilePath) {

```

Comment:
I did not like the getEnum name. I am not sure that `exploreSimulation` is the best name at the moment
Suggestion welcome!

Reply:
`getSimulationTree`?

Reply:
Either what @PavelBal suggested or `getSimulationPathsTree`